src/hera/workflows/_runner/util.py

```diff
@@ -234,9 +238,9 @@ def _runner(entrypoint: str, template_inputs_list: List) -> Any:
     # not a pydantic model with smart_union enabled
     _pydantic_mode = int(os.environ.get("hera__pydantic_mode", _PYDANTIC_VERSION))
     if _pydantic_mode == 2:
-        from pydantic import validate_call  # type: ignore
+        from pydantic import ConfigDict, validate_call  # type: ignore
 
-        function = validate_call(function)
+        function = validate_call(function, config=ConfigDict(arbitrary_types_allowed=True))  # type: ignore
```

Comment:
Does this pass type-checking if you use it like a decorator? i.e.
```suggestion
        function = validate_call(config=ConfigDict(arbitrary_types_allowed=True))(function)
```

Reply:
It does! Trying to figure out the same for the pydantic v1 branch